charts/rawls/templates/_deployment.tpl

```diff
@@ -0,0 +1,186 @@
+{{- /* Generate a Rawls deployment */ -}}
+{{- define "rawls.deployment" -}}
+{{- $settings := ._deploymentSettings -}}
+{{- $imageTag := $settings.imageTag | default .Values.global.applicationVersion -}}
+{{- $legacyResourcePrefix := $settings.legacyResourcePrefix | default $settings.name -}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ $settings.name }}-deployment
+  labels:
+{{ include "rawls.labels" . | indent 4 }}
+spec:
+  replicas: {{ $settings.replicas }}
+  revisionHistoryLimit: 0
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 0
+      maxUnavailable: 1
+  selector:
+    matchLabels:
+      deployment: {{ $settings.name}} 
+  template: 
+    metadata:
+      labels:
+        deployment: {{ $settings.name }}
+{{ include "rawls.labels" . | indent 8 }}
+    spec: 
+      serviceAccountName: rawls-sa
+      hostAliases:
+      - ip: 127.0.0.1 
+        hostnames:
+        - app
+        - sqlproxy
+      volumes:
+      - name: app-ctmpls
+        secret:
+          secretName: {{ $legacyResourcePrefix }}-app-ctmpls
+      - name: proxy-ctmpls
+        secret:
+          secretName: {{ $legacyResourcePrefix }}-proxy-ctmpls
+      - name: sqlproxy-ctmpls
+        secret:
+          secretName: {{ $legacyResourcePrefix }}-sqlproxy-ctmpls
+      - name: {{ $settings.name }}-proxy-security-logs
+        emptyDir: {}
+      - name: rawls-prometheusjmx-jar
+        emptyDir: {}
+      - name: {{ $settings.name }}-cm
+        configMap:
+          name: {{ $settings.name }}-cm
+      containers:
+      - name: {{ $settings.name }}-app
+        image: "gcr.io/broad-dsp-gcr-public/rawls:{{ $imageTag }}"
+        resources:
+{{ toYaml .Values.resources | indent 10 }}
+        ports:
+          - containerPort: 8080 
+          - name: metrics
+            containerPort: 9090
+            protocol: TCP
+        envFrom:
+        - secretRef:
+            name: {{ $legacyResourcePrefix }}-app-env
+        env: 
+        - name: K8S_NODE_NAME
+          valueFrom: 
+            fieldRef:
+              fieldPath: spec.nodeName
+        - name: K8S_POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: PROMETHEUS_ARGS
+          value: "-javaagent:/etc/prometheusjmx/prometheusjmx.jar=9090:/etc/rawls-cm/prometheusJmx.yaml"
+        command: ["/bin/bash"]
+        args:
+        - '-c'
+        - >-
+          java ${JAVA_OPTS}
+          {{- if .Values.prometheus.enabled }}
```

Comment:
IMO if we are going to have this flag we should make the chart work without Prometheus if it is disabled. See [this commit](https://github.com/broadinstitute/terra-helm/pull/182/commits/1f02162a633c959fe7c484183254f74045dc11b2) for an example of the changes needed.

Reply:
I'm not sure this flag is necessary. I can't think of a situation where we'd want to set it to false off the top of my head. I actually went through and removed the flag, but I missed this one. 

I think if we ever do have a use case for this flag its easy enough to go in and add it.